### PR TITLE
ISPN-7342 - Cassandra Cache Store is missing Custom Cache Store service manifest

### PR DESCRIPTION
**Pull request by nmvijay**
https://issues.jboss.org/browse/ISPN-7342

Created proper /META-INF/services/org.infinispan.persistence.spi.AdvancedLoadWriteStore entry pointing to CassandraStore class.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,78 +1,119 @@
...
-      <artifactId>infinispan-persistence-parent</artifactId>
-      <version>9.0.0-SNAPSHOT</version>
-   </parent>
-   <artifactId>infinispan-cachestore-cassandra</artifactId>
-   <packaging>bundle</packaging>
-   <name>Infinispan CassandraCacheStore</name>
-   <description>Infinispan CassandraCacheStore module</description>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
```
Comment (slaskawi) [inline pom.xml]: Could you please reformat it along with our [Infinispan formatter](https://github.com/infinispan/infinispan/tree/master/ide-settings)?

File: pom.xml [hunk 2]
```diff
@@ -1,78 +1,119 @@
...
-                  <artifactId>guava</artifactId>
-              </exclusion>
-              <exclusion>
-                  <groupId>io.netty</groupId>
-                  <artifactId>netty-all</artifactId>
-              </exclusion>
-          </exclusions>
-      </dependency>
+	<properties>
+		<checkstyle.skip>true</checkstyle.skip>
```
Comment (slaskawi) [inline pom.xml]: Why are we skipping checkstyle?

File: pom.xml [hunk 3]
```diff
@@ -1,78 +1,119 @@
...
-      </plugins>
-   </build>
+		<dependency>
+			<groupId>org.kohsuke.metainf-services</groupId>
+			<artifactId>metainf-services</artifactId>
+			<version>1.5</version>
+			<optional>true</optional>
+		</dependency>
+
+		<!-- https://mvnrepository.com/artifact/org.testng/testng -->
```
Comment (slaskawi) [inline pom.xml]: This comment line could be removed

File: pom.xml [hunk 4]
```diff
@@ -1,78 +1,119 @@
...
+		<dependency>
+			<groupId>org.testng</groupId>
+			<artifactId>testng</artifactId>
+			<version>6.10</version>
+		</dependency>
+
+		<!-- https://mvnrepository.com/artifact/org.mockito/mockito-all -->
```
Comment (slaskawi) [inline pom.xml]: This comment line could be removed

File: pom.xml [hunk 5]
```diff
@@ -1,78 +1,119 @@
...
+			<artifactId>mockito-all</artifactId>
+			<version>1.10.19</version>
+		</dependency>
+
+	</dependencies>
+
+	<build>
+		<!-- testResources>
+			<testResource>
+				<directory>src/test/resources</directory>
```
Comment (slaskawi) [inline pom.xml]: Please remove commented code.

File: src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser [hunk 6]
```diff
@@ -1 +0,0 @@
-org.infinispan.persistence.cassandra.configuration.CassandraStoreConfigurationParser82
```
Comment (slaskawi) [inline src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser]: Why did you remove the parser?

File: pom.xml (hunk 1 above)
Comment (nmvijay) [inline pom.xml]: Sure

**Comment by nmvijay** [comment]
Please check now, I have reverted few changes made as per my project requirement.

**Comment by nmvijay** [comment]
Please check now, I have reverted few changes made as per my project requirement.

File: pom.xml (hunk 1 above)
Comment (nmvijay) [inline pom.xml]: added dependency for testng to fix compilation issue

File: src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser (hunk 6 above)
Comment (nmvijay) [inline src/main/resources/META-INF/services/org.infinispan.configuration.parsing.ConfigurationParser]: Reverted changes, please review

File: pom.xml (hunk 5 above)
Comment (nmvijay) [inline pom.xml]: Reverted changes, please review

File: pom.xml [hunk 7]
```diff
@@ -52,6 +52,13 @@
            <version>1.5</version>
            <optional>true</optional>
        </dependency>
+
+      <dependency>
+         <groupId>org.testng</groupId>
+         <artifactId>testng</artifactId>
```
Comment (slaskawi) [inline pom.xml]: This seems to be unneeded.

Comment (slaskawi) [inline pom.xml]: And also, there is no need to change permissions of `org.infinispan.configuration.parsing.ConfigurationParser` file.

Comment (slaskawi) [inline pom.xml]: Ha! Now I see it... TestNG is required...

**Comment by slaskawi** [comment]
Integrated:

- https://github.com/infinispan/infinispan-cachestore-cassandra/commit/cc30fcee1d9afd5b52abe0a28e8e525a9a5d3d3f
https://github.com/infinispan/infinispan-cachestore-cassandra/commit/4b182d4c0e4d9b5d160897b898c7c4c6b20dc2c7

Thanks @nmvijay !
